Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -83,19 +85,23 @@ ComponentInfo::componentsInSubtree(const size_t componentIndex) const {
   }
   // Calculate index into our ranges (non-detector) component items.
   const auto rangesIndex = compOffsetIndex(componentIndex);
-  const auto range = (*m_componentRanges)[rangesIndex];
+  const auto detRange = (*m_detectorRanges)[rangesIndex];
+  const auto compRange = (*m_componentRanges)[rangesIndex];
 
   // Extract as a block
-  return std::vector<size_t>(
-      m_assemblySortedComponentIndices->begin() + range.first,
-      m_assemblySortedComponentIndices->begin() + range.second);
+  std::vector<size_t> indices(
```

Comment:
This looks more complex. Not sure how this would be quicker either since the `m_assemblySortedComponentIndices` has ALL component indices already?